### PR TITLE
[Xamarin.Android.Build.Tasks] redistribute `proguard-android.txt` from Android gradle plugin

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Context: https://discord.com/channels/732297728826277939/732297837953679412/1303383272918618163

A customer found their app grew in size when building on Windows
versus macOS.

The difference being a file that was passed to the `<R8/>` task on
Windows, but not on macOS:

    C:\Users\[username]\AppData\Local\Android\Sdk\tools\proguard\proguard-android.txt

`tools` is *ancient*, as any Android SDK should be using
`cmdline-tools` now.

Let's stop using this file as a default set of proguard rules.

But we need to use the *right* set of proguard rules, reading
the text of the old one:

> This file is no longer maintained and is not used by new (2.2+)
> versions of the # Android plugin for Gradle. Instead, the Android
> plugin for Gradle generates the # default rules at build time and
> stores them in the build directory.

I found the source code for the gradle task that generates the new
files at:

* https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/ProguardFiles.java;l=116

In order to get this file, we can setup an empty gradle project
using the Android gradle plugin:

    plugins {
        id 'com.android.application' version '8.7.0'
    }

Which, hopefully, dependabot is able to update this version.

Then we can run the task:

    .\gradlew extractProguardFiles

This outputs files such as:

    src\proguard-android\build\intermediates\default_proguard_files\global\proguard-android-optimize.txt-8.7.0
    src\proguard-android\build\intermediates\default_proguard_files\global\proguard-android.txt-8.7.0
    src\proguard-android\build\intermediates\default_proguard_files\global\proguard-defaults.txt-8.7.0

For now, I simply redistributed the `proguard-android.txt` file, but
we could consider making `proguard-android-optimize.txt` useable in
the future. I believe the only difference is the
`-allowaccessmodification` flag, which users could put in their own
proguard rules.

### CODE REVIEW

**Review by dellis1972 — APPROVED** [review]

**Comment by jonathanpeppers** [comment]
Removing the line in this PR is fine for main, but it seems like the rules here might be needed:
* https://cs.android.com/android/platform/superproject/main/+/main:sdk/files/proguard-android.txt

As mentioned here:
* https://discord.com/channels/732297728826277939/732297837953679412/1303622786199064577

The Android SDK doesn't seem to redistribute any proguard rules anymore, but they have it in their repo?

**Comment by jonathanpeppers** [comment]
Hmm:
* https://stackoverflow.com/a/34554172

> This file is no longer maintained and is not used by new (2.2+) versions of the # Android plugin for Gradle. Instead, the Android plugin for Gradle generates the # default rules at build time and stores them in the build directory.

So, we might have to dig up what the Android Gradle plugin does and port the logic to C#?

**Comment by jonpryor** [comment]
We should not merge this PR until better understand the Android Gradle plugin behavior.

**Comment by jonathanpeppers** [comment]
I think I should:
- [ ] Setup a gradle project that brings in the Android gradle plugin
- [ ] Make sure dependabot can bump the Android plugin, so we can keep it updated
- [ ] Build the project with our product and ship `proguard-android.txt`
- [ ] The removed line in this PR uses the new `proguard-android.txt` file instead

Some links:
* https://cs.android.com/android/platform/superproject/main/+/main:sdk/files/proguard-android-optimize.txt
* https://maven.google.com/com/android/tools/build/gradle/8.7.2/gradle-8.7.2.jar
* https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/ProguardFiles.java;l=116

File: src/proguard-android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+repositories {
+    google()
+    mavenCentral()
+}
+
+android {
+    namespace 'com.microsoft.proguard.android'
+    // Setting the minimum we support at the moment, might not matter
+    compileSdk 21
```
Comment (dellis1972) [inline src/proguard-android/build.gradle:13]: aren't we on 23 now?

Comment (jonathanpeppers) [reply src/proguard-android/build.gradle:13]: The runtime is still 21 right now, but I don't think their gradle task even looks at this number:

* https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/ProguardFiles.java;l=116

I get a build error if you don't list some number, though:

```
A problem occurred configuring root project 'proguard-android'.
> compileSdkVersion is not specified. Please add it to build.gradle
```

**Review by dellis1972 — APPROVED** [review]

**Comment by jonathanpeppers** [comment]
I was going to add a TPN, like:
```csharp
using System;
using System.Collections.Generic;
using System.IO;

namespace Xamarin.Android.Prepare;

[TPN]
class android_build_tools_gradle_TPN : ThirdPartyNotice
{
	// https://mvnrepository.com/artifact/com.android.tools.build/gradle
	static readonly Uri    url         = new Uri ("https://android.googlesource.com/platform/tools/base");

	public override string LicenseFile => CommonLicenses.Apache20Path;
	public override string Name        => "google/com.android.tools.build/gradle";
	public override Uri    SourceUrl   => url;
	public override string LicenseText => String.Empty;

	public override bool   Include (bool includeExternalDeps, bool includeBuildDeps) => includeExternalDeps;
}
```

When running `-t:Prepare` I saw no changes?!?

Then I found we already have an entry for this:

https://github.com/dotnet/android/blob/70948d5eaa72091389940fdd5a0eb828c85bdd60/build-tools/xaprepare/xaprepare/ThirdPartyNotices/Xamarin.Android.Build.Tasks.cs#L7-L16

So, maybe we are good and no TPN changed needed?

**Comment by jonpryor** [comment]
Should we update the existing TPN URL to? https://android.googlesource.com/platform/tools/base/+/refs/heads/main/sdk-common/NOTICE

(explicit branch name of `master` vs. explicit commit)

**Review by grendello — APPROVED** [review]

**Pull request merged**